src/transformers/models/bridgetower/modeling_bridgetower.py

```diff
@@ -0,0 +1,1731 @@
+                [
+                    ("c_fc", nn.Linear(d_model, d_model * 4)),
+                    ("gelu", QuickGELUActivation()),
+                    ("c_proj", nn.Linear(d_model * 4, d_model)),
+                ]
+            )
+        )
+        self.ln_2 = BridgeTowerLayerNorm(d_model)
+        self.attn_mask = attn_mask
+
+    def attention(self, hidden_state: torch.Tensor, attention_mask: torch.Tensor):
+        if attention_mask is not None:
+            attention_mask = attention_mask.to(dtype=torch.bool, device=hidden_state.device)
+        self.attn_mask = (
+            self.attn_mask.to(dtype=hidden_state.dtype, device=hidden_state.device)
+            if self.attn_mask is not None
+            else None
+        )
+        return self.attn(
+            hidden_state,
+            hidden_state,
+            hidden_state,
+            need_weights=False,
+            attn_mask=self.attn_mask,
+            key_padding_mask=attention_mask,
+        )[0]
+
+    def forward(self, hidden_state: torch.Tensor, attention_mask: torch.Tensor = None):
+        residual_state = hidden_state + self.attention(self.ln_1(hidden_state), attention_mask)
+        hidden_state = self.ln_2(residual_state)
+        for _, layer in self.mlp.items():
+            hidden_state = layer(hidden_state)
+        hidden_state = residual_state + hidden_state
+        return hidden_state
+
+
+class BridgeTowerTransformer(nn.Module):
+    def __init__(
+        self,
+        hidden_size: int,
+        num_hidden_layers: int,
+        heads: int,
+        attn_mask: torch.Tensor = None,
+        model_type: str = "bridgetower",
+        stop_gradient: bool = False,
+        remove_last_layer: bool = False,
+    ):
+        super().__init__()
+        self.hidden_size = hidden_size
+        self.num_hidden_layers = num_hidden_layers
+        if remove_last_layer:
+            self.resblocks = nn.ModuleList(
+                [BridgeTowerResidualAttention(hidden_size, heads, attn_mask) for _ in range(num_hidden_layers - 1)]
+            )
+        else:
+            self.resblocks = nn.ModuleList(
+                [BridgeTowerResidualAttention(hidden_size, heads, attn_mask) for _ in range(num_hidden_layers)]
+            )
+        self.model_type = model_type
+        self.stop_gradient = stop_gradient
+
+    def forward(self, hidden_state: torch.Tensor, attention_mask: torch.Tensor = None):
+        hidden_states = []
+        for block in self.resblocks:
+            hidden_state = block(hidden_state, attention_mask)
+            if self.model_type == "bridgetower":
+                if self.stop_gradient:
+                    hidden_states.append(hidden_state.detach())
+                else:
+                    hidden_states.append(hidden_state)
+        if self.model_type == "bridgetower":
+            return hidden_states
+        else:
+            return hidden_state
+
+
+class BridgeTowerVisualTransformer(nn.Module):
+    def __init__(
+        self,
```

Comment:
This should take the config of the model, not individual arguments.

Reply:
We have fixed this in our latest commit. Thanks for the feedback!